haskellPackages: update stackage and hackage

cdepillabout opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/Mw5WLnzsP7fC4Zky) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #151578. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

sternenseemann committed:
haskell.compiler: ghc901 -> ghc902

https://www.haskell.org/ghc/download_ghc_9_0_2.html

sternenseemann committed:
haskellPackages: allow 9.0.2 versions for the ghc-lib* packages

These updated constraints will take effect the next time the hackage
data tarball is updated and the 9.0.2.* versions become available.

sternenseemann committed:
all-cabal-hashes: 2021-12-21T13:58:08Z -> 2021-12-27T15:07:56Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: stackage-lts 18.19 -> 18.20

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

sternenseemann committed:
Merge #152136: haskell.compiler: ghc901 -> ghc902

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

sternenseemann committed:
haskell.packages.ghc902: adjust pinned ghc-lib* versions

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
all-cabal-hashes: 2021-12-27T15:07:56Z -> 2021-12-28T00:33:48Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

cdepillabout committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

cdepillabout committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

sternenseemann committed:
haskell.packages.ghc902.hie-bios: patch for breaking change in ghc

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell.compiler.*: disable large address space only on iOS

The condition used in the past to detect iOS was "is this
aarch64-darwin"? Since we have aarch64-darwin devices running macOS
nowadays which do allow large address space, let's use the more accurate
flag.

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout commented:
There are currently a whole bunch of failures on Darwin:

https://github.com/cdepillabout/nix-haskell-updates-status/tree/ed66299bcbcff2122c7abef32b37b50c37685c85

On Matrix, sterni was suggesting that these errors were introduced somewhere between 44f52594d1a0dda1848f227b34a0b088dcb1e11d and c066e2f31649da0324f3e9a2f60cf9834e2261db.

I'd appreciate help from anyone with interest in OSX in tracking this down and fixing it.

Some people that I've seen on past OSX-related issues that might be interested: @domenkozar, @roberth, @jkachmar, @NixOS/darwin-maintainers 

--------------------------------

Not strictly related, but there are also a few darwin-related Haskell PRs currently open that could use a review or a final push:

- https://github.com/NixOS/nixpkgs/pull/149942
- https://github.com/NixOS/nixpkgs/pull/152037
- https://github.com/NixOS/nixpkgs/pull/152581 (this one is probably fine and just needs to be merged in, although it would of course be welcome if someone could test)
- https://github.com/NixOS/nixpkgs/pull/147964

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout commented:
@veprbl posted another likely darwin-related issue at https://github.com/NixOS/nixpkgs/issues/152859.

sternenseemann commented:
> @veprbl posted another likely darwin-related issue at #152859.

Not another, that is the exact issue.

sternenseemann committed:
haskellPackages.http-client-tls: force rebuild on darwin

Currently everything seems like the derivation's outPath has been
corrupted somehow, causing reverse dependencies to fail. Since we can't
purge store paths from Hydra and the binary cache easily at the moment,
we just make the derivation hash change by inserting an unsubstantial
change, causing Hydra to build this package and all reverse dependencies
again — hopefully successfully this time around.

arcz committed:
hevm: unbreak on aarch64-linux

(cherry picked from commit 1468f4479bee1b334515be92f5e5f7032c71ba08)

This is from https://github.com/NixOS/nixpkgs/pull/153344.

sternenseemann committed:
haskellPackages.foldl: force rebuild on darwin

See comment and message for 2341c7e4853.

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.gtk2hs-buildtools: remove -O0 workaround

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh

cdepillabout committed:
haskellPackages.crackNum: re-enable on aarch64-linux

This should hopefully now be working on aarch64-linux since
sbv (one of crackNum's deps) is now working on aarch64-linux:
https://github.com/NixOS/nixpkgs/pull/153344
